packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/TableDemo/TableEditableDemo.tsx

```diff
@@ -0,0 +1,304 @@
+import * as React from 'react';
+import { TextInput } from '@patternfly/react-core';
+import {
+  Table,
+  TableHeader,
+  TableBody,
+  TableProps,
+  TableVariant,
+  ICell,
+  IRow,
+  IRowCell,
+  IValidatorDef,
+  OnRowEdit,
+  validateCellEdits,
+  cancelCellEdits,
+  applyCellEdits
+} from '@patternfly/react-table';
+
+const rowLevelValidationRules: IValidatorDef[] = [
+  {
+    name: 'required',
+    validator: value => value.trim() !== '',
+    errorText: 'This field is required'
+  },
+  {
+    name: 'notFoo',
+    validator: value => value.trim().toLowerCase() !== 'foo',
+    errorText: 'Value cannot be "foo"'
+  },
+  {
+    name: 'minLength',
+    validator: value => value.trim().length >= 7,
+    errorText: 'Value must be at least 7 characters'
+  },
+  {
+    name: 'notXyz',
+    validator: value => value.trim().toLowerCase() !== 'xyz',
+    errorText: 'Value cannot be xyz'
+  }
+];
+
+export class TableEditableDemo extends React.Component<TableProps, { columns: (ICell | string)[]; rows: IRow[] }> {
+  constructor(props: TableProps) {
+    super(props);
+    this.state = {
+      columns: [
+        'Text input col 1',
+        'Disabled text input col 2',
+        'Text input col 3',
+        'Text input col 4'
+      ],
+      rows: [
+        {
+          isEditable: true,
+          rowEditValidationRules: rowLevelValidationRules,
+          cells: [
+            {
+              title: (value: string, rowIndex: number, cellIndex: number, props: any) => (
+                <React.Fragment>
+                  <div className="pf-c-inline-edit__value">{value}</div>
```

Comment:
Why are you adding css into the integration app?

Reply:
The markup used here in react-integration was out of date, have since replaced with a wrapper component `EditableTextCell` to make it easier to work with and less error-prone for users. Is it ok?